### PR TITLE
raftstore: get read index independently in batch

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/7971

Problem Summary:
We use the same index to respond to the batch of `read index` responses which breaks the correctness.
See more details in https://github.com/tikv/tikv/issues/7971.

### What is changed and how it works?

What's Changed:
Starting from the second, all the following read index requests will get its own committed index.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

Side effects

No.

### Release note 
* Fix the issue that a read index request may get out-of-date committed index.

### CODE REVIEW

File: components/raftstore/src/store/peer.rs [hunk 1]
```diff
@@ -2117,10 +2122,11 @@ impl Peer {
                 // before or after the previous read index, and the lease can be renewed when get
                 // heartbeat responses.
                 LeaseState::Valid | LeaseState::Expired => {
+                    let committed_index = self.get_store().committed_index();
                     if let Some(read) = self.pending_reads.back_mut() {
                         let max_lease = poll_ctx.cfg.raft_store_max_leader_lease();
                         if read.renew_lease_time + max_lease > renew_lease_time {
-                            read.push_command(req, cb);
+                            read.push_command(req, cb, committed_index);
```
Comment (hicqu) [inline components/raftstore/src/store/peer.rs:2129]: Suppose there are 2 read index requests A and B, and A reaches the TiKV peer earlier. Seems B can get a less index than A. Is it correct?

Comment (gengliqi) [reply components/raftstore/src/store/peer.rs:2129]: It doesn't matter. These two operations are parallel so the order is not guaranteed.

**Pull request merged**